### PR TITLE
Simplified selectExistingPaths

### PR DESCRIPTION
**Pull request by sakhuja**
1. selectExistingPaths method simplification. 
2. handling for order free path validity in the path argument.

**Related issues**
[Issues]https://github.com/salesforce/TransmogrifAI/issues/483

**Describe the proposed solution**
In selectExistingPaths method, we check for the component paths validity using FileSystem object's read outcome. Also, make use of scala's Try method to bucketize valid and invalid paths.
as part of this change we also make sure that the order of the component paths is not critical to the success of the selectExistingPaths method, specially when the first path component is invalid. Added a test as well to verify that we can handle order independent paths.

### CODE REVIEW

**Comment by sakhuja** [comment]
@gerashegalov This change is ready for review. 

please note: I've added my salesforce email id to my account to confirm with salesforce-cla requirement.

File: utils/src/main/scala/com/salesforce/op/utils/io/avro/AvroInOut.scala [hunk 1]
```diff
@@ -86,18 +86,26 @@ object AvroInOut {
...
-      val separator = FileSystems.getDefault.getSeparator
-      val firstPath = paths.head.split(separator)
-      val bucket = firstPath.slice(0, math.min(4, firstPath.length)).mkString(separator)
-      FileSystem.get(new URI(bucket), sc.sparkContext.hadoopConfiguration)
-    } catch {
-      case ex: Exception => throw new IllegalArgumentException(s"Bad path '$path': ${ex.getMessage}")
+    val paths = path.split(',')
+    val fsSeq = paths.map(path => Try(FileSystem.get(new URI(path), sc.sparkContext.hadoopConfiguration)))
+    val validFS = fsSeq.filter(p => p.isSuccess)
+    if (validFS.nonEmpty) {
```
[the remark this answers is not in the document]
Comment (gerashegalov) [reply utils/src/main/scala/com/salesforce/op/utils/io/avro/AvroInOut.scala:92]: I was hoping to keep the code short to the tune of 
```scala
    val paths = Option(path).map(_.split(',')).getOrElse(Array.empty).map(new Path(_))
    val found = paths.headOption.flatMap(firstPath => 
      Try(firstPath.getFileSystem(sc.sparkContext.hadoopConfiguration)).recover { case t => 
        throw new IllegalArgumentException(s"Bad path '$path': ${t.getMessage}", t)
      }.toOption)
      .map(fs => paths.filter(fs.exists)).getOrElse(Array.empty)
```

File: utils/src/main/scala/com/salesforce/op/utils/io/avro/AvroInOut.scala [hunk 2]
```diff
@@ -86,17 +84,13 @@ object AvroInOut {
...
-    }
-    val found = paths.filter(p => fs.exists(new Path(p)))
-    if (found.isEmpty) throw new IllegalArgumentException(s"No valid directory found in path '$path'")
+    val paths = Option(path).map(_.split(',')).getOrElse(Array.empty).map(new Path(_))
+    val found = paths.headOption.flatMap(firstPath =>
+      Try(firstPath.getFileSystem(sc.sparkContext.hadoopConfiguration)).recover { case t =>
+        throw new IllegalArgumentException(s"Bad path '$path': ${t.getMessage}", t)
+      }.toOption)
+      .map(fs => paths.filter(fs.exists)).getOrElse(Array.empty)
+    if(found.isEmpty) throw new IllegalArgumentException(s"No valid directory found in path '$path'")
```
Comment (gerashegalov) [inline utils/src/main/scala/com/salesforce/op/utils/io/avro/AvroInOut.scala:93]: nit: whitespace disappeared in `if (found`

File: utils/src/test/scala/com/salesforce/op/utils/io/avro/AvroInOutTest.scala [hunk 3]
```diff
@@ -109,13 +109,22 @@ class AvroInOutTest extends FlatSpec with TestSparkContext {
     f3.delete()
     assume(f1.exists && !f2.exists && !f3.exists)
 
+    // check for one dir being invalid in the path amongst two
```
Comment (gerashegalov) [inline utils/src/test/scala/com/salesforce/op/utils/io/avro/AvroInOutTest.scala:112]: thanks for comments

**Comment by sakhuja** [comment]
Thanks for the review @gerashegalov. Could you help merge? I don't have write-access.

**Pull request merged**